### PR TITLE
Add post-training quantization and mixed bit-width quantization

### PR DESCRIPTION
**Pull request by linbinskn**

### CODE REVIEW

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 1]
```diff
@@ -110,12 +112,111 @@ def get_bits_length(config, quant_type):
     else:
         return config["quant_bits"].get(quant_type)
 
+def _check_weight(module):
+    try:
+        return isinstance(module.weight.data, torch.Tensor)
+    except AttributeError:
+        return False
+
+class Post_Training_Quantize():
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:121]: the coding style should be consistent with existing code. For example, `Quantize` -> `Quantizer`, `Post_Training_Quantize` -> `PostTrainingQuantize`

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 2]
```diff
@@ -110,12 +112,111 @@ def get_bits_length(config, quant_type):
...
+        rmax = torch.max(weight)
+        rmin = torch.min(weight)
+        scale, zero_point = update_quantization_param(quant_bits, rmin, rmax)
+        return scale, zero_point
+
+    def quantize(self):
+        """
+        quantize real value to specific bit-width.
+
+        Parameters
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:162]: if there is no parameter, just remove it

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 3]
```diff
@@ -110,12 +112,111 @@ def get_bits_length(config, quant_type):
...
+        ----------
+        None
+
+        Returns
+        -------
+        torch.nn.module
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:168]: `torch.nn.module` -> `torch.nn.Module`. and explain the returned model a little bit

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 4]
```diff
@@ -7,14 +7,15 @@
 from ..utils.config_validation import CompressorSchema
 from ..compressor import Quantizer, QuantGrad, QuantType
 
-__all__ = ['NaiveQuantizer', 'QAT_Quantizer', 'DoReFaQuantizer', 'BNNQuantizer']
+__all__ = ['NaiveQuantizer', 'QAT_Quantizer', 'DoReFaQuantizer', 'BNNQuantizer', 'Post_Training_Quantize']
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:10]: please also help change `QAT_Quantizer` to `QATQuantizer`

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 5]
```diff
@@ -110,12 +112,111 @@ def get_bits_length(config, quant_type):
...
+
+class Post_Training_Quantize():
+    """Post_Training_Quantize contains two major function. The first is global scale quantization.
+    The second is mixed scale quantization.
+    """
+
+    def __init__(self, model, quant_bits=-1):
+        self.model = model
+        self.quant_bits = quant_bits
+        """
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:129]: it is really strange that you put docstring behind the implementation of `__init__`

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py [hunk 6]
```diff
@@ -110,12 +112,111 @@ def get_bits_length(config, quant_type):
...
 
+def _check_weight(module):
+    try:
+        return isinstance(module.weight.data, torch.Tensor)
+    except AttributeError:
+        return False
+
+class Post_Training_Quantize():
+    """Post_Training_Quantize contains two major function. The first is global scale quantization.
+    The second is mixed scale quantization.
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:123]: it is not clear that what is the first and what is second

File: src/sdk/pynni/nni/compression/torch/quantization/quantizers.py (hunk 1 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:121]: it is also strange that it does not inherit the class `Quantizer`

Comment (linbinskn) [reply src/sdk/pynni/nni/compression/torch/quantization/quantizers.py:121]: I think the implementation of class `Quantizer` is designed for 'quantize aware training'. It is not suitable for post-training quantization and mixed bit-width quantization.

**Pull request closed without merging**